src/database/index.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseApp } from '../firebase-app';
+import * as firebaseRtdbApi from '@firebase/database';
+import * as firebaseRtdbTypesApi from '@firebase/database-types';
+import * as adminRtdbApi from './database';
+
+export function database(app: FirebaseApp): firebaseRtdbApi.Database {
+  return app.database();
+}
+
+// We must define a namespace to make the typings work correctly. 
+// Otherwise `admin.database()` cannot be called like a function.
+/* eslint-disable @typescript-eslint/no-namespace */
+export namespace admin.database {
```

Comment:
Shouldn't this be just `export namespace database`? Or was this done as another temporary measure since we don't yet have the top-level barrel export?

Reply:
Yep, the reason I did this is because we cannot perform a top-level barrel export yet and inside the existing `src/index.d.ts` file it expects an `admin.database` instead of simply `database`. If I change it to just `database`, then we'd need to change the `src/index.d.ts` as well to point to the correct namespace. 

I didn't want to change it for existing code as it reads from `src/database.d.ts`, though I considered adding a `.replace()` in the gulpfile for our tests to replace the code of `src/index.d.ts` at build time but it felt less clean than this solution.

So for example today's `src/index.d.ts` looks like this:

```
declare namespace admin.database {
  export import Database = _database.admin.database.Database;
  // this would have to be _database.database.Database; but right now it's _database.admin.database.Database
  // ... plus eventually we will just barrel export anyway
}
```